samcli/commands/validate/validate.py

```diff
@@ -95,3 +108,63 @@ def _read_sam_file(template):
         sam_template = yaml_parse(sam_template.read())
 
     return sam_template
+
+
+def _lint(ctx: Context, template: str) -> None:
+    """
+    Parses provided SAM template and maps errors from CloudFormation template back to SAM template.
+
+    Cfn-lint loggers are added to the SAM cli logging hierarchy which at the root logger
+    configures with INFO level logging and a different formatting. This exposes and duplicates
+    some cfn-lint logs that are not typically shown to customers. Explicitly setting the level to
+    WARNING and propagate to be False remediates these issues.
+
+    Parameters
+    -----------
+    ctx
+        Click context object
+    template
+        Path to the template file
+
+    """
+
+    import cfnlint.core  # type: ignore
+    import logging
+    from samcli.commands.exceptions import UserException
+    from botocore.utils import validate_region_name
+    from botocore.exceptions import InvalidRegionError
+
+    LOGGER = logging.getLogger("cfnlint")
+    LOGGER.propagate = False
+
+    try:
+        validate_region_name(ctx.region)
```

Comment:
I think this is good (I didn’t know such function exist) but I believe we should be doing this higher level, instead of manually checking it.

Reply:
by higher level do you mean within the do_cli method? That could be better, as I believe running "sam validate --region --debug" results in printing the stack trace

Reply:
I agree with @mndeveci , we should move this functionality and makes it so that its commonly used across all commands within SAM CLI.